scripts/pre_commit_linter.py

```diff
@@ -714,11 +718,142 @@ def _check_spacing(all_files):
+        line_num = 0
+        while line_num < length:
+            line = content[line_num].lstrip().rstrip()
+            if line.startswith('#'):
+                if line_num + 1 == length:
+                    if line[-1] not in PUNCTUATIONS and line[1:].lstrip()[0].isupper():
+                        failed = True
+                        print '%s --> Line %s: %s' % (
+                            filename, line_num + 1, message)
+                    break                        
+
+                next_line = content[line_num + 1].lstrip().rstrip()
+
+                if not next_line.startswith('#'):
+                    if line[-1] not in PUNCTUATIONS and line[1:].lstrip()[0].isupper():
+                        failed = True
+                        print '%s --> Line %s: %s' % (
+                            filename, line_num + 1, message)
+
+                else:
+                    while line_num + 1 < length and next_line.startswith('#'):
+                        line_num += 1
+                        next_line = content[line_num].lstrip().rstrip()
+                    if next_line.startswith('#'):
+                        line = next_line
+                        line_num += 1
+                    else:
+                        line = content[line_num - 1].lstrip().rstrip()
+                    if (
+                        line[-1] not in PUNCTUATIONS and
+                        not any(word in line
+                        for word in EXCLUDED_WORDS)):
+                        failed = True
+                        print '%s --> Line %s: %s' % (
+                            filename, line_num, message)
+
+            line_num += 1
+
+    print ''
+    print '----------------------------------------'
+    print ''
+    if failed:
+        summary_message = (
+            '%s   Comments check failed' % _MESSAGE_TYPE_FAILED)
+        print summary_message
+        summary_messages.append(summary_message)
+    else:
+        summary_message = (
+            '%s   Comments check passed' % _MESSAGE_TYPE_SUCCESS)
+        print summary_message
+        summary_messages.append(summary_message)
+
+    return summary_messages
+
+
+def _check_docstrings(all_files):
+    """This functions checks for a period at the end of docstring."""
+    print 'Starting docstring checks'
+    print '----------------------------------------'
+    summary_messages = []
+    files_to_check = [
+        filename for filename in all_files if not
+        any(fnmatch.fnmatch(filename, pattern) for pattern in EXCLUDED_PATHS)
+        and filename.endswith('.py')]
+    message = 'There should be a period at the end of the docstring.'
+    failed = False
+    for filename in files_to_check:
+        f = open(filename, 'r')
+        content = f.readlines()
+        length = len(content)
+        line_num = 0
+        while line_num < length:
+            line = content[line_num].lstrip().rstrip()
+
+            # Check for single line docstring.
+            if (
```

Comment:
Remove parenthesis around the predicate.

Reply:
Can you please make this clear why this is needed ? I don't think we need to do that.

Reply:
It's for styling, less so for function.